src/libfetchers/git-utils.cc

```diff
@@ -644,17 +660,112 @@ struct GitInputAccessor : InputAccessor
     }
 };
 
-ref<InputAccessor> GitRepoImpl::getAccessor(const Hash & rev)
+struct GitExportIgnoreInputAccessor : FilteringInputAccessor {
+    ref<GitRepoImpl> repo;
+    std::optional<Hash> rev;
+
+    GitExportIgnoreInputAccessor(ref<GitRepoImpl> repo, ref<InputAccessor> next, std::optional<Hash> rev)
+        : FilteringInputAccessor(next, [&](const CanonPath & path) {
+            return RestrictedPathError(fmt("'%s' does not exist because it was fetched with exportIgnore enabled", path));
+        })
+        , repo(repo)
+        , rev(rev)
+    { }
+
+    bool gitAttrGet(const CanonPath & path, const char * attrName, const char * & valueOut) {
+        std::string pathStr {path.rel()};
+        const char * pathCStr = pathStr.c_str();
```

Comment:
Better to add a `rel_c_str()` to `CanonPath`. I.e.
```
    const char * raw_c_str() const
    { return path.c_str() + 1; }
```

Reply:
Done.